### PR TITLE
Define color for a merchant while registering.

### PR DESCRIPTION
**Pull request by tomschw**
This PR will enable merchants to chose their own color or to be able to set a defined color for a merchant.

### CODE REVIEW

File: api/marketplace.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def holding_cost_rate(self, merchant_id: str) -> float:
         response = self.request('get', 'holding_cost_rate/' + merchant_id)
         return response.json()
 
-    def register(self, endpoint_url_or_port: Union[str, int], merchant_name: str,
+    def register(self, endpoint_url_or_port: Union[str, int], merchant_name: str, color: str,
```
Comment (Bouncner) [inline api/marketplace.py:47]: What happens if color not set? Have you thought about a boring default such as '#gray'?

File: merchant.py [hunk 2]
```diff
@@ -13,6 +13,9 @@ class CheapestStrategy:
         'default_price': 30
     }
 
+    # color can be set and will always be the same, otherwise random color from predefined set
```
Comment (Bouncner) [inline merchant.py]: > color can be set and will always be the same

What does that mean?

File: merchant.py [hunk 3]
```diff
@@ -35,6 +38,8 @@ class TwoBoundStrategy:
         'lower_price_bound': 20
     }
 
+    color = "#33a02c"
```
Comment (Bouncner) [inline merchant.py:43]: Can you add a comment what these colors are?

File: pricewars_merchant.py [hunk 4]
```diff
@@ -7,6 +7,7 @@
 import random
 from typing import Optional, List
 from pathlib import Path
+import random
```
Comment (Bouncner) [inline pricewars_merchant.py]: Line 7?

File: pricewars_merchant.py [hunk 5]
```diff
@@ -17,7 +18,11 @@ class PricewarsMerchant(metaclass=ABCMeta):
     # Save/Read token file in merchant directory
     TOKEN_FILE = Path(__file__).parent / 'auth_tokens.json'
 
-    def __init__(self, port: int, token: Optional[str], marketplace_url: str, producer_url: str, merchant_name: str):
+    
+
+    def __init__(self, port: int, token: Optional[str], marketplace_url: str, producer_url: str, merchant_name: str, color: Optional[str]):
```
Comment (Bouncner) [inline pricewars_merchant.py:20]: Why to much whitespace?

File: pricewars_merchant.py [hunk 6]
```diff
@@ -17,7 +18,11 @@ class PricewarsMerchant(metaclass=ABCMeta):
...
+        colors = ['#a6cee3','#1f78b4','#b2df8a','#fb9a99','#fdbf6f','#ff7f00','#cab2d6','#6a3d9a']
```
Comment (Bouncner) [inline pricewars_merchant.py:23]: Color brewer? If so, add comment.

Comment (Bouncner) [reply pricewars_merchant.py:23]: In any way ... add comment. What does this array do?

File: pricewars_merchant.py [hunk 7]
```diff
@@ -33,6 +38,12 @@ def __init__(self, port: int, token: Optional[str], marketplace_url: str, produc
         self.number_offered_items = 0
         self.products_not_offered = []
 
+        if not color:
+            random.seed(time.time())
```
Comment (Bouncner) [inline pricewars_merchant.py]: Seed not needed.

**Review by Bouncner — CHANGES_REQUESTED** [review]

File: api/marketplace.py (hunk 1 above)
Comment (tomschw) [reply api/marketplace.py:47]: If the pricers_merchant is used as a framework we will always have a defined color. Also if others use this function, they will know that register needs a color as input, so I don't see any reason for a default.

Comment (Bouncner) [reply api/marketplace.py:47]: Makes sense.

File: merchant.py (hunk 2 above)
Comment (tomschw) [reply merchant.py]: @Bouncner Adjusted comments.

File: pricewars_merchant.py [hunk 8]
```diff
@@ -17,7 +17,11 @@ class PricewarsMerchant(metaclass=ABCMeta):
     # Save/Read token file in merchant directory
     TOKEN_FILE = Path(__file__).parent / 'auth_tokens.json'
 
-    def __init__(self, port: int, token: Optional[str], marketplace_url: str, producer_url: str, merchant_name: str):
+    def __init__(self, port: int, token: Optional[str], marketplace_url: str, producer_url: str, merchant_name: str, color: Optional[str]):
+        # default colors which are assigned to a merchant without it's own color
```
Comment (Bouncner) [inline pricewars_merchant.py]: its

Comment (Bouncner) [reply pricewars_merchant.py]: For multiple sentences, end a sentence with a period.

File: pricewars_merchant.py [hunk 9]
```diff
@@ -33,6 +37,11 @@ def __init__(self, port: int, token: Optional[str], marketplace_url: str, produc
         self.number_offered_items = 0
         self.products_not_offered = []
 
+        if not color:
+            color = colors[random.randint(0, len(colors)-1)]
+
+        self.color = color
+
```
Comment (Bouncner) [inline pricewars_merchant.py:43]: You use too much whitespace for my taste. No need to change that here. But these lines belong together as they all handle the colors. Why separate them?

File: pricewars_merchant.py (hunk 8 above)
Comment (tomschw) [reply pricewars_merchant.py]: Done.

File: pricewars_merchant.py (hunk 9 above)
Comment (tomschw) [reply pricewars_merchant.py:43]: I don't know why, but I would rather use too much whitespace than too little. But you are right, this one is unnecessary.

**Review by Bouncner — APPROVED** [review]

**Pull request merged**